### PR TITLE
fix(errors): Update errorBoundaryLoader behaviour

### PR DESCRIPTION
**Pull request by srascar**
* improve status code management
* improve error stack trace management
* add tests

Solves https://github.com/ekino/rendr/issues/29

### CODE REVIEW

File: packages/loader/src/index.ts [hunk 1]
```diff
@@ -28,17 +28,48 @@ export const createChainedLoader: (loaders: Loader[]) => Loader = loaders => {
...
+// 3. The resultPage receives the message of the error in the key message of its settings
+// 4. The context receives the error trace in the key stackTrace of its settings
+// 5. In non production mode, resultPage also receives the error trace
 export const errorBoundaryLoader: Loader = async (ctx, page, next) => {
   let resultPage: Page | void;
   try {
     resultPage = await next();
   } catch (err) {
+    if (ctx.res.headersSent) {
+      ctx.res.end();
```
Comment (rande) [inline packages/loader/src/index.ts]: I think we need a `createErrorBoundaryLoader` an optional logger option. or at least with a callback

File: packages/loader/src/index.ts [hunk 2]
```diff
@@ -28,17 +28,48 @@ export const createChainedLoader: (loaders: Loader[]) => Loader = loaders => {
...
+
     resultPage = createPage();
     resultPage.statusCode = err instanceof NotFoundError ? 404 : 500;
+
+    if (ctx.res.statusCode >= 300) {
+      resultPage.statusCode = ctx.res.statusCode;
+    }
+
+    const fullStack = flattenStackTrace(err);
+    ctx.settings.stackTrace = fullStack;
```
Comment (rande) [inline packages/loader/src/index.ts]: We need to put this as a block `rendr.text` ?

File: packages/loader/src/index.ts [hunk 3]
```diff
@@ -28,17 +28,48 @@ export const createChainedLoader: (loaders: Loader[]) => Loader = loaders => {
...
     resultPage.settings.message = err.message;
+
+    if (ctx.isServerSide && process.env.NODE_ENV !== "production") {
+      resultPage.settings.stackTrace = fullStack;
```
Comment (rande) [inline packages/loader/src/index.ts]: We need to call a callback here, we cannot have silent error with no log.

File: packages/loader/src/index.ts (hunk 1 above)
Comment (srascar) [reply packages/loader/src/index.ts]: done

**Comment by rande** [comment]
@srascar can you resolve this conflict on this PR ? also I have removed the reference on the variable `errorBoundaryLoader` to prefer a function call: `createErrorBoundary` so all helper functions as the same naming convention.

Thanks!

File: packages/loader/src/index.ts (hunk 1 above)
Comment (rande) [reply packages/loader/src/index.ts]: now we have the logger, can we log to it the error ?

File: packages/loader/src/index.ts [hunk 4]
```diff
@@ -28,20 +35,66 @@ export const createChainedLoader: (loaders: Loader[]) => Loader = loaders => {
...
+      return handler(err, ctx, page);
     }
-
-    return resultPage;
   };
-};
+}
+
+// Simple version of the errorBoundaryLoader that uses console as a logger
+export const errorBoundaryLoader = createErrorBoundaryLoader();
```
Comment (rande) [inline packages/loader/src/index.ts]: can we remove the export of the variable ? let's keep the `createSomething` convention.

**Pull request merged**